src/app/layout.tsx

```diff
@@ -19,9 +18,7 @@ export default function RootLayout({
     <html lang="en">
       <body className={inter.className} id="root">
         <div id="__next">
-          <UserSessionContextProvider>
-            <ThemeProviderWrapper>{children}</ThemeProviderWrapper>
-          </UserSessionContextProvider>
+          <ThemeProviderWrapper>{children}</ThemeProviderWrapper>
```

Comment:
Looking at https://zustand.docs.pmnd.rs/guides/nextjs#app-router
We need to provide the stores through contexts because of

> No global stores - Because the store should not be shared across requests, it should not be defined as a global variable. Instead, the store should be created per request.

Reply:
Also, Zustand's error handling here reminded me of Next.js's error handling. It might be cleaner to just throw the error in our (Store initialization and Context) and fallback onto an error.tsx.  

Right now, OfficeHourState has an Error / Message state and I assume we were planning to conditionally render? Am I missing another usecase for the Error/Message state. 
@nickbar01234 what do you think?


> export const useCounterStore = <T,>(
>   selector: (store: CounterStore) => T,
> ): T => {
>   const counterStoreContext = useContext(CounterStoreContext)
> 
>   if (!counterStoreContext) {
>     throw new Error(`useCounterStore must be used within CounterStoreProvider`)
>   }
> 
>   return useStore(counterStoreContext, selector)
> }